### PR TITLE
Implement Edit meeting feature

### PR DESCRIPTION
**Pull request by tomascherian**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        parseTagsForEdit(argMultiMap.getAllValues(PREFIX_TAG)).ifPresent(editMeetingDescriptor::setTags);
+        parseParticipantsIndexForEdit(argMultiMap.getAllValues(PREFIX_PARTICIPANTS))
+                .ifPresent((editMeetingDescriptor::setParticipantsIndex));
+
+        if (!editMeetingDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EditContactCommand.MESSAGE_NOT_EDITED);
+        }
+        return new EditMeetingCommand(index, editMeetingDescriptor);
+    }
```
Comment (cpwill01) [inline src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java:71]: This method is a bit long, maybe you might want to look into shortening it when improving code quality in v1.4

**Review by cpwill01 — APPROVED** [review]
Seems to be working, LGTM

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+        assert meetingToEdit != null;
+
+        MeetingName updatedMeetingName = editMeetingDescriptor.getMeetingName().orElse(meetingToEdit.getName());
+        MeetingDate updatedMeetingDate = editMeetingDescriptor.getMeetingDate().orElse(meetingToEdit.getDate());
+        StartTime updatedStartTime = editMeetingDescriptor.getStartTime().orElse(meetingToEdit.getStartTime());
+        EndTime updatedEndTime = editMeetingDescriptor.getEndTime().orElse(meetingToEdit.getEndTime());
+
+        if (updatedStartTime.isAfter(updatedEndTime)) {
+            throw new CommandException(MESSAGE_INVALID_TIME);
+        }
```
Comment (YiFei2) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:113]: I think for this part we should use `!updatedStartTime.isBefore(updatedEndTime)` or `updatedStartTime.isAfter(updatedEndTime) && updatedStartTime == updatedEndTime` to throw exception for the case  where startTime is equal to endTime!

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+            for (Index targetIndex : editMeetingDescriptor.participantsIndex) {
+                if (targetIndex.getZeroBased() >= contactList.size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+                }
+                Contact participatingContact = contactList.get(targetIndex.getZeroBased());
+                updatedParticipants.add(new Participant(participatingContact));
+            }
+        } else {
+            updatedParticipants.addAll(meetingToEdit.getParticipants());
+        }
```
Comment (YiFei2) [inline src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:130]: Just to note that there is some deep nesting here. I think we can work on it later on if there's no easy abstraction to fix it now.

File: src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>} if {@code tags} is non-empty.
+     * If {@code tags} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Tag>} containing zero tags.
+     */
```
Comment (YiFei2) [inline src/main/java/seedu/address/logic/parser/EditMeetingCommandParser.java:77]: Could possibly just do a similar write up for parseParticipantsIndexForEdit as well!

File: src/main/java/seedu/address/logic/commands/EditMeetingCommand.java (hunk 2 above)
Comment (tomascherian) [reply src/main/java/seedu/address/logic/commands/EditMeetingCommand.java:113]:
```
public boolean isAfter(Time time) {
        return !(this.value.compareTo(time.value) < 0);
 }
```
Actually `Time::isAfter` takes into account the `updatedStartTime == updatedEndTime` case

**Pull request merged**